stdlib/result.gr

```diff
@@ -1,89 +1,203 @@
-export let isOk = (v) => {
-    match (v) {
-        Ok(_)  => true,
-        _      => false
-    }
+/**
+ * @module Result: Utilities for working with the Result data type.
+ * 
+ * The Result type is an enum that represents the possibility of a success case (with the `Ok` variant),
+ * or an error case (with the `Err` variant). Use a Result as the return type of a function that may return an error.
+ * 
+ * @example import Result from "result"
+ * 
+ * 
+ * @example let success = Ok("Yay!") // Creates a successful Result containing "Yay!"
```

Comment:
nitpick - if we made the Ok value something other than a string, it would be more obvious it doesn't have to be a string like Err

Reply:
I agree. Of alternatively add more examples there.